tests/tensor/test_variable.py

```diff
@@ -98,10 +100,15 @@ def test_infix_matmul_method():
     assert equal_computations([res], [exp_res])
 
     X_val = np.arange(2 * 3).reshape((2, 3))
-    res = as_tensor(X_val) @ y
+    res = X_val @ y
```

Comment:
This is the test relevant to the reported bug

Reply:
I was running `test_dot_method`, and it failed with an error that `__dot__` was not implemented

Reply:
Right but if that's the only test that used it's not a reason to have the mysterious dunder method. I couldn't see it being used anywhere else notlr could I see how to trigger it naturally